packages/overlay/src/popper.ts

```diff
@@ -25,6 +25,35 @@ import preventOverflow from '@popperjs/core/lib/modifiers/preventOverflow';
 import arrow from '@popperjs/core/lib/modifiers/arrow';
 import offset from '@popperjs/core/lib/modifiers/offset';
 import { computeArrowRotateStyles } from './popper-arrow-rotate';
+import maxSize from 'popper-max-size-modifier';
+
+const applyMaxSize = {
+    name: 'applyMaxSize',
+    enabled: true,
+    phase: 'beforeWrite',
+    requires: ['maxSize'],
+    fn({
+        state,
+    }: {
+        state: {
+            modifiersData: {
+                maxSize: {
+                    height: number;
+                };
+            };
+            styles: {
+                popper: {
+                    maxHeight: string;
+                    height: string;
+                };
+            };
+        };
+    }) {
+        const { height } = state.modifiersData.maxSize;
+        state.styles.popper.maxHeight = `${height}px`;
+        state.styles.popper.height = `${height}px`;
```

Comment:
I think this is causing the behavior that @cuberoot noted.

```suggestion
```

Reply:
100%, yes. It's a bit of a bag of hacks right now to offset some weird CSS intersections that don't make sense to me right now, which is why this is for discussion _only_.

Is there a way to cache the _first_ `const { height: actualHeight } = state.rects.popper;` internal to popper, or would we need to do that ourselves in this modifier?

Reply:
In popper we use to put state we want to persist across updates in `state.modifiersData['modifierName#persistent']`.

https://github.com/popperjs/popper-core/blob/bf9d83dd46aa059234e59f26b6a6362999b93937/src/index.js#L201-L204

It's just a convention, actually any modifier state named after something not listed in the modifiers list is ignored, but this may change in the future, it's just an implementation detail.